src/utils/fhirConversionUtils.ts

```diff
@@ -0,0 +1,207 @@
+import { CodeableConcept, Condition, MedicationStatement, Observation, Procedure } from 'fhir/r4';
+import { fhirclient } from 'fhirclient/lib/types';
+
+type FhirUser = fhirclient.FHIR.Patient | fhirclient.FHIR.Practitioner | fhirclient.FHIR.RelatedPerson;
+
+type FhirUserName = {
+  prefix: string;
+  given: string[];
+  family: string;
+  suffix?: string[];
+};
+
+export type Patient = {
+  id: string;
+  name: string;
+  gender: string;
+  age: string;
+  zipcode: string;
+};
+
+export type PrimaryCancerCondition = {
+  cancerType: string;
+  cancerSubtype: string;
+  stage: string;
+};
+
+export type User = {
+  id: string;
+  name: string;
+  record?: FhirUser; // for debugging
+};
+
+export const convertFhirKarnofskyPerformanceStatus = (bundle: fhirclient.FHIR.Bundle): string => {
+  const loincToKarnofskyMap = {
+    'LA29175-9': '100',
+    'LA29176-7': '90',
+    'LA29177-5': '80',
+    'LA29178-3': '70',
+    'LA29179-1': '60',
+    'LA29180-9': '50',
+    'LA29181-7': '40',
+    'LA29182-5': '30',
+    'LA29183-3': '20',
+    'LA29184-1': '10',
+    'LA9627-6': '0',
+  };
+
+  const observation = bundle.entry?.[0]?.resource as Observation;
+  const code = observation?.interpretation?.[0]?.coding?.[0]?.code;
+  return loincToKarnofskyMap[code] ?? null;
+};
+
+export const convertFhirEcogPerformanceStatus = (bundle: fhirclient.FHIR.Bundle): string => {
+  const loincToEcogMap = {
+    'LA9622-7': '0',
+    'LA9623-5': '1',
+    'LA9624-3': '2',
+    'LA9625-0': '3',
+    'LA9626-8': '4',
+    'LA9627-6': '5',
+  };
+
+  const observation = bundle.entry?.[0]?.resource as Observation;
+  const code = observation?.interpretation?.[0]?.coding?.[0]?.code;
+  return loincToEcogMap[code] ?? null;
+};
+
+export const convertFhirMedicationStatements = (bundle: fhirclient.FHIR.Bundle): string[] => {
+  const medicationStatements = bundle.entry?.map(entry => entry.resource as MedicationStatement) || [];
+  const medications: string[][] = medicationStatements.map(getDisplays);
+  return getUniques(medications.flat());
+};
+
+export const convertFhirPatient = (fhirPatient: fhirclient.FHIR.Patient): Patient => ({
+  id: fhirPatient.id,
+  name: `${fhirPatient.name[0].given[0]} ${fhirPatient.name[0].family}`,
+  gender: fhirPatient.gender,
+  age: fhirPatient.birthDate ? getAge(fhirPatient.birthDate) : null,
+  zipcode:
+    fhirPatient.address?.length > 0 && fhirPatient.address[0].postalCode ? fhirPatient.address[0].postalCode : null,
+});
+
+export const convertFhirPrimaryCancerCondition = (bundle: fhirclient.FHIR.Bundle): PrimaryCancerCondition => {
+  const condition = bundle?.entry?.[0]?.resource as Condition;
+  return {
+    cancerType: condition?.bodySite?.[0]?.coding?.[0]?.display ?? null,
+    cancerSubtype: condition?.code?.coding?.[0]?.display ?? null,
```

Comment:
The cancer type should be the `Condition.code`. The `Condition.bodySite` code, if present, would indicate where the condition is located. From the FHIR R4 spec on the `bodySite` property: "Only used if not implicit in code found in Condition.code."

From the MCode spec it looks like the exact cancer type is a combination of the `Condition.code` and `Condition.bodySite` codes. However, most of the codes we have seem to imply a body site. For example, for "Breast Cancer," we have the SNOMED code "372137005 - Primary malignant neoplasm of breast (disorder)."

I don't know exactly what the subtype should be here. The example I have uses the [MCode histology morphology behavior extension](https://hl7.org/fhir/us/mcode/StructureDefinition-mcode-histology-morphology-behavior.html) which I guess could be a "subtype"?

So fixing this isn't quite as simple as using the `Condition.code` for the type and the `Condition.bodySite` for the subtype. It may involve some more clarification on what these fields are.